src/target/riscv/riscv-013.c

```diff
@@ -1592,7 +1607,8 @@ static int deassert_reset(struct target *target)
 		if (target->reset_halt) {
 			LOG_DEBUG("Waiting for hart %d to halt out of reset.", index);
 			do {
-				if (dmstatus_read(target, &dmstatus, true) != ERROR_OK)
+				if (dmstatus_read_timeout(target, &dmstatus, true,
+							riscv_reset_timeout_sec) != ERROR_OK)
 					return ERROR_FAIL;
 				if (time(NULL) - start > riscv_reset_timeout_sec) {
```

Comment:
You'll basically never get this error message in the case of a timeout, because you'll hit the timeout in the `dmstatus_read_timeout`, print an unhelpful message for the user, and never print the `increase riscv_reset_timeout_sec` message.